addons/arsenal/functions/fnc_scanConfig.sqf

```diff
@@ -127,20 +127,18 @@ private _magazineGroups = [[],[]] call CBA_fnc_hashCreate;
     };
 } foreach configProperties [_configCfgWeapons, "isClass _x && {(if (isNumber (_x >> 'scopeArsenal')) then {getNumber (_x >> 'scopeArsenal')} else {getNumber (_x >> 'scope')}) == 2} && {getNumber (_x >> 'ace_arsenal_hide') != 1}", true];
 
+private _grenadeList = [];
 {
-    private _className = configName _x;
-    private _grenadeList = [];
-    {
-        _grenadeList append getArray (_configCfgWeapons >> "Throw" >> _x >> "magazines");
-        false
-    } count getArray (_configCfgWeapons >> "Throw" >> "muzzles");
+    _grenadeList append getArray (_configCfgWeapons >> "Throw" >> _x >> "magazines");
+} foreach getArray (_configCfgWeapons >> "Throw" >> "muzzles");
 
-    private _putList = [];
-    {
-        _putList append getArray (_configCfgWeapons >> "Put" >> _x >> "magazines");
-        false
-    } count getArray (_configCfgWeapons >> "Put" >> "muzzles");
+private _putList = [];
+{
+    _putList append getArray (_configCfgWeapons >> "Put" >> _x >> "magazines");
+} foreach getArray (_configCfgWeapons >> "Put" >> "muzzles");
```

Comment:
How about removing duplicates with arrayIntersect? 
Won't be that big of a difference as the lookup is done in-engine anyway

Reply:
is it even possible to have duplicates in there?

Reply:
Nooo Idea. But I'd say bad Modders can do anything?

Reply:
Why should we be fixing what bad modders do then? Show duplicates, so they are more likely to remove them even.

Reply:
It wouldn't even matter if there were duplicates, since you're just checking below if the classname is present in the list